STIR now has linking errors when using parallelproj

One of the recent changes has broken something as we now have
https://github.com/SyneRBI/SIRF-SuperBuild/runs/2313382843?check_suite_focus=true#step:6:13025
```
/home/runner/install/lib/librecon_buildblock.a(ForwardProjectorByBinParallelproj.cxx.o): In function `stir::ForwardProjectorByBinParallelproj::set_input(stir::DiscretisedDensity<3, float> const&)':
ForwardProjectorByBinParallelproj.cxx:(.text+0xf94): undefined reference to `joseph3d_fwd(float const*, float const*, float const*, float const*, float const*, float*, long long, int const*)'
```

--- a/c/include/parallelproj_c.h
+++ b/c/include/parallelproj_c.h
@@ -5,6 +5,10 @@
 #ifndef __PARALLELPROJ_C_H__
 #define __PARALLELPROJ_C_H__
 
+#ifdef  __cplusplus
+extern "C" {
+#endif
+
 /** @brief 3D non-tof joseph back projector
  *
  *  All threads back project in one image using openmp's atomic add.
@@ -326,4 +330,5 @@ void joseph3d_fwd_tof_sino(const float *xstart,
                            float n_sigmas,
                            short n_tofbins);
 
+}  /* extern "C" */
 #endif

